serveradmin/api/decorators.py

```diff
@@ -77,19 +84,22 @@ def _wrapper(request):
             if isinstance(error, ObjectDoesNotExist):
                 status_code = 404
 
+            logger.error('{} request data: {}'.format(str(error), {
+                'now': now,
+                'body': body,
+                'public_keys': public_keys,
+                'signatures': signatures,
+                'app_id': app_id,
+                'token': token,
+                'then': then,
+                'body_json': body_json,
+            }), exc_info=True)
```

Comment:
> Things logged here aren't serveradmin errors but client errors
I am pretty sure we have a backend error here as we can reproduce the error by changing the backend code.

> Also this will likely log a huge amount of data when things go south and that data will not be readily understandable by humans either
I disagree with that. The change actually helped a lot in debugging the error and making sure there was no wrong request data sent to the backend.

> If any you should bring back something like logger.error('api: Forbidden: ' + problem) here.
I think we are almost there. How about we take your suggestion to log the message as you suggested and log the request data on level debug which allows the person running the servers to log it to a custom location if needed?

Reply:
> we explicitly switched away from this approach to raising Exceptions instead

Do you have any details why we decided to go for this approach. The problem I have here is that I get bug reports from users not getting to much details about why the 403 happened and I have no way to prove that it is client side.

Reply:
To give you a bit more context about what we are talking. This is a exception we are facing without the change but not anymore with the change (I removed the data part as it contains sensitive data but I was able to redo the request with the request data successfully):

`
[2020-03-27 20:23:27] serveradmin._wrapper:97 ERROR Disabled application request data: [...]
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/serveradmin/api/decorators.py", line 64, in _wrapper
    public_keys, signatures, app_id, token, then, now, body
  File "/usr/lib/python3/dist-packages/serveradmin/api/decorators.py", line 151, in authenticate_app
    raise PermissionDenied('Disabled application')
django.core.exceptions.PermissionDenied: Disabled application
`

It raises the exception here: https://github.com/innogames/serveradmin/blob/master/serveradmin/api/decorators.py#L140

but it is completely dubious to me how. I even installed a trigger on the database and no application was disabled at any time while this error occurred. Unfortunately I don't see how the changes related can this to be fixed.

Reply:
See https://github.com/innogames/serveradmin/pull/83#issuecomment-609846007